libr/core/disasm.c

```diff
@@ -4166,7 +4170,17 @@ static void ds_print_esil_anal(RDisasmState *ds) {
 	}
 	switch (ds->analop.type) {
 	case R_ANAL_OP_TYPE_SWI: {
-		char *s = cmd_syscall_dostr (core, -1);
+		char *s;
+		if (disasm_isThumb (core)) {
+			RAnalOp aop = {0};
+			int bsize = 64;
+			ut8 buf[bsize];
+			r_io_read_at_mapped (core->io, at, buf, bsize);
+			r_anal_op (core->anal, &aop, core->offset, buf, bsize, R_ANAL_OP_MASK_ESIL);
+			s = cmd_syscall_dostr (core, aop.val);
```

Comment:
Move the logic inside cmdsyscalldo

Reply:
I would need an 'at' address inside cmd_syscall_dostr to be able to do that.